snapshotter/demux/snapshotter_test.go

```diff
@@ -73,15 +73,36 @@ func TestReturnErrorWhenCalledWithoutNamespacedContext(t *testing.T) {
 		{"View", func() error { _, err := uut.View(ctx, "layerKey", ""); return err }},
 		{"Commit", func() error { return uut.Commit(ctx, "layer1", "layerKey") }},
 		{"Remove", func() error { return uut.Remove(ctx, "layerKey") }},
+	}
+
+	for _, test := range tests {
+		if err := test.run(); err == nil {
+			t.Fatalf("%s call did not return error", test.name)
+		}
+	}
+}
```

Comment:
Do any of these tests validate return values, or do they simply confirm that there were no errors?  Is output validation handled elsewhere?

I guess that's a moot point for this PR though, since `WalkAll` only returns the `error` type...

Reply:
In their current form, they do not. Maybe room for improvement. The idea was to test expected behaviors under specific conditions. The errors themselves are mocks.